modules/renter/upload_test.go

```diff
@@ -7,47 +7,37 @@ import (
 	"time"
 
 	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/crypto"
 	"github.com/NebulousLabs/Sia/modules"
-	"github.com/NebulousLabs/Sia/modules/renter/hostdb"
+	"github.com/NebulousLabs/Sia/modules/renter/contractor"
 	"github.com/NebulousLabs/Sia/types"
 )
 
-// uploadHostDB is a mocked hostDB, hostdb.HostPool, and hostdb.Uploader. It
-// is used for testing the uploading and repairing functions of the renter.
-type uploadHostDB struct{}
+// uploadContractor is a mocked hostContractor and contractor.Editor. It is
+// used for testing the uploading and repairing functions of the renter.
+type uploadContractor struct {
+	stubContractor
+}
 
-// NewPool returns a new mock HostPool. Since uploadHostDB implements the
-// HostPool interface, it can simply return itself.
-func (hdb uploadHostDB) NewPool(uint64, types.BlockHeight) (hostdb.HostPool, error) {
-	return hdb, nil
+func (uploadContractor) Contracts() []contractor.Contract {
+	return make([]contractor.Contract, 24) // exact number shouldn't matter, as long as its large enough
```

Comment:
I am in favor of using a larger number than the max number of hosts typically chosen. Especially as we move down the line, having 30 hosts instead of 24 means that the renter can start making decisions like "choose 3 really cheap but unreliable hosts instead of 1 good but really expensive host". I realized it's actually the wrong strategy to pick a redundancy, instead you should pick a reliability (such as eleven 9's) and let the renter figure out the cheapest/best combination of hosts to hit that reliability. Which means you may need more than 6x redundancy, or may mean you only need 2 redundancy, just depends on how reliable vs. expensive the hosts are.